tests/FSharp.Compiler.ComponentTests/Language/PrintfFormatTests.fs

```diff
@@ -61,4 +63,32 @@ let test = sprintf Format 42
     'string'    
 but here has type
     'int'    "
-        }
\ No newline at end of file
+        }
+
+[<Fact>]
+let ``Non-inline literals cannot be used as printf format in lang version70``() =
+    let csLib = 
+        CSharp """
+public static class Library
+{
+    public const string Version = "1.0.0";
+}"""    |> withName "CsLib"
+
+    FSharp """
+module PrintfFormatTests
+
+[<Literal>]
+let Format = "%s%d%s"
+
+let bad1 = sprintf Format "yup" Format.Length (string Format.Length)
+let ok1 = sprintf "%s" Format
+let bad2 = sprintf Library.Version
+    """
+    |> withLangVersion70
+    |> withReferences [csLib]
+    |> compile
+    |> shouldFail
+    |> withDiagnostics [
+        (Error 3350, Line 7, Col 20, Line 7, Col 26, "Feature 'String values marked as literals and IL constants as printf format' is not available in F# 7.0. Please use language version 'PREVIEW' or greater.")
+        (Error 3350, Line 9, Col 20, Line 9, Col 35, "Feature 'String values marked as literals and IL constants as printf format' is not available in F# 7.0. Please use language version 'PREVIEW' or greater.")
```

Comment:
Some nitpicking here: technically in this context there is nothing "greater" than preview :) at some point (after the next F# release) this message will become misleading.

C# has these things like "Feature 'file-scoped namespace' is not available in C# 7.3. Please use language version 10.0 or greater."

Reply:
3350 is an existing diagnostic though.

Reply:
Right. Okay, nevermind then. @vzarytovskii probably we should run through these cases in the codebase, these messages are not future-proof :)